crates/rethnet_evm/benches/util.rs

```diff
@@ -0,0 +1,125 @@
+use std::clone::Clone;
+
+use criterion::{BatchSize, BenchmarkId, Criterion};
+use rethnet_eth::{Address, Bytes, U256};
+use rethnet_evm::state::{HybridState, LayeredState, RethnetLayer, StateError, SyncState};
+use revm::primitives::{AccountInfo, Bytecode};
+
+#[derive(Default)]
+struct RethnetStates {
+    layered: LayeredState<RethnetLayer>,
+    hybrid: HybridState<RethnetLayer>,
+}
+
+impl RethnetStates {
+    fn fill(&mut self, number_of_accounts: u64, number_of_accounts_per_checkpoint: u64) {
+        let mut states: [&mut dyn SyncState<StateError>; 2] = [&mut self.layered, &mut self.hybrid];
+        for state in states.iter_mut() {
+            let number_of_checkpoints = number_of_accounts / number_of_accounts_per_checkpoint;
+            for checkpoint_number in 0..number_of_checkpoints {
+                for account_number in 1..=number_of_accounts_per_checkpoint {
+                    let account_number =
+                        (checkpoint_number * number_of_accounts_per_checkpoint) + account_number;
+                    let address = Address::from_low_u64_ne(account_number);
+                    state
+                        .insert_account(
+                            address,
+                            AccountInfo::new(
+                                U256::from(account_number),
+                                account_number,
+                                Bytecode::new_raw(Bytes::copy_from_slice(address.as_bytes())),
+                            ),
+                        )
+                        .unwrap();
+                    state
```

Comment:
Addressed in https://github.com/NomicFoundation/hardhat/pull/3910/commits/dd5e0f81c5a69edcf658a33a22f38da3ce9b731a

Reply:
Could you please explain your design choice for the enum to vary storage slots in the API?

It's a different choice than I had considered, so curious to hear your thoughts. I'm not thinking about the boolean vs enum discussion btw; I'm happy with the enum.

Reply:
I guess I didn't make any choice between different tradeoffs; I just did what felt expedient. Which part of the code are you referring to? How would you have done it differently?

Reply:
To me, it seemed like the part of the benchmark that's variable is the storage scales, so I would just simple have exposed that in the api: `storage_scales: &[u64]` and at the call site would pass in `&STORAGE_SCALES` or `&[1]` if you don't want to test multiple scales.

It's explicit and prevents needing to add additional types like `VaryStorageSlots`.

Hence, why I was curious what your train of thought was

Reply:
IMO, using the enum actually makes it feel more explicit rather than less, since when you're reading the call, you can see what the parameter is, which neither a boolean nor a `&[1]` gives you. And, forcing the caller to be concerned the caller with the `*_SCALES` feels like it adds at least as much complexity as the enum, if not more. But I don't feel that strongly about it, so I went ahead and changed it as you suggested, in https://github.com/NomicFoundation/hardhat/pull/3910/commits/d3d22b6ce4af43f8ccd3aac3129c7ee085ea73b9. Let me know if that's what you expected.